YellowNotes.Core/Repositories/CategoryRepository.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using YellowNotes.Core.Dtos;
+using YellowNotes.Core.Models;
+
+namespace YellowNotes.Core.Repositories
+{
+    public class CategoryRepository : ICategoryRepository
```

Comment:
this should be only db access, without any logic. all logic, business etc should goes to service.